test/sqllogictest/transform/is_null_propagation.slt

```diff
@@ -50,62 +50,57 @@ EXPLAIN SELECT FROM t1 WHERE f2 IN ( SELECT agg1 FROM ( SELECT COUNT ( TRUE ) ag
 | Get materialize.public.t1 (u1)
 | Distinct group=(#1)
 
-%1 =
-| Get materialize.public.t2 (u3)
-| ArrangeBy ()
+%1 = Let l1 =
+| Get materialize.public.t1 (u1)
+| Project (#1)
 
 %2 =
-| Get %0 (l0)
+| Get materialize.public.t2 (u3)
+| Project (#1)
 | ArrangeBy ()
 
 %3 =
-| Get materialize.public.t1 (u1)
+| Get %0 (l0)
 | ArrangeBy ()
 
 %4 =
 | Get materialize.public.t1 (u1)
+| Project ()
+| ArrangeBy ()
 
-%5 = Let l1 =
-| Join %1 %2 %3 %4
-| | implementation = Differential %4 %1.() %2.() %3.()
-| | demand = (#1, #2, #6)
-| Filter (#6 > #1)
-| Reduce group=(#2)
+%5 = Let l2 =
+| Join %2 %3 %4 %1
+| | implementation = Differential %1 %2.() %3.() %4.()
+| Filter (#2 > #0)
```

Comment:
It would be great if there were a `Project` here, as this communicates to the join that once the filter is applied it can drop .. well pretty much all of the input other than `#1` (the only thing `Reduce` requires).